docs/user/reference/resource-compute/list-of-commands-available-on-a-compute-resource-provisioned-by-juju/list-of-juju-introspect-macros/juju_agent.md

```diff
@@ -1,4 +1,55 @@
 (juju_agent)=
 # `juju_agent`
 
-TBA
+The juju_agent script is a proxy function which 
+will pass the arguments to the correct agent (
+machine, controller, application or unit).
+It is used to simplify the usage of the juju-introspect.
+
+## Usage
+
+```text
+juju_agent [command]                                  
+```
+
+## Details
+
+juju_agent is one of the shell functions that 
+is exported by the juju-introspect shell script. 
+This subcommand is simply a selector for the  
+correct agent to use when passing another subcommand
+to juju-instrospect.
+
+By default, juju_agent will transfer the subcommand  
+to the machine agent. 
+If the machine agent is not running (i.e. if the 
+agent name is null), then it tries to transfer 
+to the controller agent. If the controller agent is 
+not running, then it tries to transfer to the application 
+agent. And, finally, if the application agent is not 
+running, then it tries to transfer to the unit agent.
+
+## Example 
+```code
+juju_agent metrics
+```
+_Note: This example is equivalent to running juju_metrics._
+
```

Comment:
It would be good to find a different example that shows the unique value of the juju_agent macro.